### PR TITLE
test: cleaning up some H3 TODOs

### PR DESCRIPTION
**Pull request by alyssawilk**
If compiled out, H3 is automatically removed from getProtocolTestParams here
https://github.com/envoyproxy/envoy/blob/main/test/integration/http_protocol_integration.cc#L14

Risk Level: n/a (test only)
Testing: manual
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

File: test/integration/protocol_integration_test.cc [hunk 1]
```diff
@@ -2016,13 +2008,9 @@ name: encode-headers-return-stop-all-filter
...
   ASSERT_TRUE(response->complete());
   // Data is added in encodeData for all protocols, and encodeTrailers for HTTP/2 and above.
-  int times_added = (upstreamProtocol() == FakeHttpConnection::Type::HTTP1 ||
-                     downstreamProtocol() == Http::CodecClient::Type::HTTP1)
-                        ? 1
-                        : 2;
+  int times_added = upstreamProtocol() == FakeHttpConnection::Type::HTTP1 ? 1 : 2;
   if (downstreamProtocol() == Http::CodecClient::Type::HTTP1 &&
       upstreamProtocol() == FakeHttpConnection::Type::HTTP3) {
-    // TODO(alyssawilk) Figure out why the bytes mismatch with the test expectation below.
```
Comment (asraa) [inline test/integration/protocol_integration_test.cc:2025]: is this byte mismatch expected for H/3 upstream?

**Review by asraa — COMMENTED** [review]
Thanks!

**Comment by alyssawilk** [comment]
oops - ran the wrong tests locally, this needs fixing.
/wait

**Comment by alyssawilk** [comment]
ah, nm, unrelated failure.  merged main to pick up the latest fixes.

**Review by asraa — APPROVED** [review]
LGTM after CI!

**Pull request merged**